common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -549,18 +546,59 @@ def test_user_can_revoke_and_no_refund_available(
         assert CourseEnrollment.is_enrolled(self.user, self.course.id)
 
         # Unenroll with Revoke for refund
-        with patch('lms.djangoapps.commerce.signals.handle_refund_entitlement') as mock_refund_handler:
-            REFUND_ENTITLEMENT.connect(mock_refund_handler)
-
-            revoke_url = url + '?is_refund=true'
-            response = self.client.delete(
-                revoke_url,
-                content_type='application/json',
-            )
-            assert response.status_code == 400
-
-            course_entitlement.refresh_from_db()
-            assert not mock_refund_handler.called
-            assert CourseEnrollment.is_enrolled(self.user, self.course.id)
-            assert course_entitlement.enrollment_course_run is not None
-            assert course_entitlement.expired_at is None
+        revoke_url = url + '?is_refund=true'
+        response = self.client.delete(
+            revoke_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 400
```

Comment:
I think you changed this to 500

Reply:
NVM, I misunderstood this test.

I don't understand why it returns a 400 though? Shouldn't it succeed at removing the enrollment, but not call `refund_entitlement` (since `is_entitlement_refundable` is `False`)?

Reply:
```
        else:
            log.info(
                'Entitlement Refund failed for Course Entitlement [%s]. Entitlement is not refundable',
                entitlement.uuid
            )
            return Response(
                status=status.HTTP_400_BAD_REQUEST,
                data={
                    'message': 'Entitlement refund failed, Entitlement is not refundable'
                })
```

We explicitly added this state in the earlier review of this code.  I thought it was part of the review at least.  

It implies that if you are trying to refund an entitlement that is not refundable the operation should fail.

I can check with Deen also.

Reply:
Sorry, I think I may have misread this. I think its good the way it is.